crates/astria-composer/src/executor/mod.rs

```diff
@@ -250,8 +255,8 @@ impl Executor {
                     }
                 }
 
-                // receive new seq_action and bundle it
-                Some(seq_action) = self.serialized_rollup_transactions.recv() => {
+                // receive new seq_action and bundle it. will not pull from the channel if `bundle_factory` is full
+                Some(seq_action) = self.serialized_rollup_transactions.recv(), if !bundle_factory.full() => {
                     let rollup_id = seq_action.rollup_id;
                     if let Err(e) = bundle_factory.try_push(seq_action) {
                             warn!(
```

Comment:
this condition will run before `Some(seq_action)` is pulled from the channel. this means that the channel won't be pulled on while `bundle_factory` is full. this will cause `try_send` calls in the collectors to timeout, and that's where transactions will be dropped.

Reply:
ah got it. that makes sense.